src/Injector.js

```diff
@@ -14,94 +9,50 @@ export default class ReduxInjector extends Component {
     children: PropTypes.func.isRequired,
     actions: PropTypes.objectOf(
       PropTypes.func.isRequired
-    ).isRequired,
-    stores: PropTypes.objectOf(
-      PropTypes.func.isRequired
     ).isRequired
   }
 
   static defaultProps = {
-    stores: {},
     actions: {}
   };
 
   constructor(props, context) {
     super(props, context);
-    this.handleChange = this.handleChange.bind(this);
-    this.update(props);
-  }
 
-  componentWillReceiveProps(nextProps) {
-    this.update(nextProps);
+    this.handleChange = this.handleChange.bind(this);
+    this.unsubscribe = context.redux.subscribe(this.handleChange);
   }
 
   componentWillUnmount() {
     this.unsubscribe();
   }
 
-  update(props) {
-    const { stores, actions } = props;
-
-    if (process.env.NODE_ENV !== 'production') {
-      invariant(
-        isPlainObject(actions),
-        '"actions" must be a plain object with functions as values. Instead received: %s.',
-        actions
-      );
-      invariant(
-        isPlainObject(stores),
-        '"stores" must be a plain object with functions as values. Instead received: %s.',
-        stores
-      );
-      Object.keys(actions).forEach(key =>
-        invariant(
-          isFunction(actions[key]),
-          'Expected "%s" in "actions" to be a function. Instead received: %s.',
-          key,
-          actions[key]
-        )
-      );
-      Object.keys(stores).forEach(key =>
-        invariant(
-          isFunction(stores[key]),
-          'Expected "%s" in "stores" to be a function. Instead received: %s.',
-          key,
-          stores[key]
-        )
-      );
-    }
-
-    const { wrapActionCreator, observeStores } = this.context.redux;
-    this.actions = mapValues(props.actions, wrapActionCreator);
+  performAction(actionCreator, ...args) {
+    const { dispatch, atom } = this.context.redux;
+    const payload = actionCreator(...args);
 
-    if (this.unsubscribe) {
-      this.unsubscribe();
-    }
-
-    this.unsubscribe = observeStores(values(stores), this.handleChange);
-  }
-
-  mapState(stateFromStores) {
-    const { getStoreKey } = this.context.redux;
-    return mapValues(this.props.stores, store =>
-      stateFromStores[getStoreKey(store)]
-    );
+    return typeof payload === 'function'
+      ? payload(dispatch, atom)
+      : dispatch(payload);
   }
 
-  handleChange(stateFromStores) {
-    const nextState = this.mapState(stateFromStores);
+  handleChange(atom) {
     if (this.state) {
-      this.setState(nextState);
+      this.setState({ atom });
     } else {
-      this.state = nextState;
+      this.state = { atom };
     }
   }
 
   render() {
-    const { children } = this.props;
-    return children({
-      state: this.state,
-      actions: this.actions
-    });
+    const { children, actions: _actions } = this.props;
+    const { atom } = this.state;
+
+    const actions = Object.keys(_actions).reduce((result, key) => {
```

Comment:
Yeah, might use that. I kinda like the rawness though (I didn't realize I could use `reduce` like this).
I don't have strong opinion here. (Note it's also used in `combineStores`)